### ISSUE TITLE
Verify the case where k < windowsize

### ISSUE THREAD

**Issue by chatox** [fair-search/fairsearch-elasticsearch-plugin#32]
I understand in the current code the variable k can be specified independently of the window size of the re-ranker.

In particular, it must be k <= windowsize (if k > windowsize => exception).

Now, when k < windowsize, the M-Table should be applied up to position k. After that position, we do not verify if the number of protected elements is greater or equal than anything, because there is no M-Table. In other words, the M-Table applies to the first k elements, the remaining elements are returned in the order in which they come, independently on whether they are protected or not.

This issue requires:

1. @purbon to certify that indeed k can be < windowsize

2. @tsuehr to check that the re-ranker works correctly when it receives more elements than the size of the M-Table

**Comment by purbon** [comment]
Fair point @chatox , this relates to our last conversation about _k_ and _window_size_. Certainly _k_ can be either smaller or bigger than the _windows_size_  value used in the query,  as of the way the code is written right now.

But this does not limit us to fix this situation and make it better and more consistent with the experience.

Your solution works fine for me, we only need to add a new exception for this case in the reranker function.

> Now, when k < windowsize, the M-Table should be applied up to position k.

This will be always like this.

**Comment by purbon** [comment]
Now that I think more on this problem, this situation will only occur when using the fixed lookup for proportion strategy.
